credit_card_checker/credit_card_checker.rb

```diff
@@ -1,5 +1,101 @@
+require './bank_data'
+
+# Class provides methods for validation of credit card numbers based on luhn algorithm
+#   and specific bank information like IIN ranges and code lengthes.
+# Specific bank data is stored in bank_data.rb.
+
 class CreditCardChecker
+
+  # Method accepts string or int code numbers.
+  # String versions can contain '_' separators as code is converted to_i.to_s first.
+  # It returns true if card number passed luhn check and if IIN and code length
+  #   matches any registered bank in bank_data.rb.
+  # False is returned when card number fails validation or when invalid input is provided.
+
   def valid?(code)
-    'put your solution there'
+    code = validate_input(code)
+    return false if code.nil?
+    valid_luhn?(code) && valid_bank?(code[0, 6], code.size)
   end
+
+  # Method to provide more concrete information about number like why check fails -
+  #   because of a luhn algorithm failure of because bank validation.
+  # Also provides issuer name if bank check succeded.
+
+  def verbose_validation(code)
+    code = validate_input(code)
+    return {valid: false} if code.nil?
+
+    bank = bank(code[0, 6], code.size)
+    result = {
+      luhn: valid_luhn?(code),
+      issuer: bank ? bank[:issuer] : nil
+    }
+    result[:valid] = (result[:luhn] && !bank.nil?)
+    result
+  end
+
+  private
+
+    # Method assumes Int codes and String codes that looks like Int as valid.
+    # For example "123_456_789" is a valid String code.
+    # While "123_456_789a" is not.
+
+    def validate_input(code)
```

Comment:
why do you validate code for integer and string formats but don't validate for nil?